Add global styles, top navigation and footer

maribelferreira opened the pull request:
- Updated package-lock.json and package.json with new Tailwind CSS version
- Added new Skeleton.astro layout with global styles and basic page structure

# What kind of change does this PR introduce?
 
<!-- (Bug fix, feature, docs update, ...) -->

# What is the current behavior?
 
<!-- You can also link to an open issue here -->

# What is the new behavior (if this is a feature change)?

# Other information

maribelferreira committed:
Upgrade Tailwind CSS and related dependencies to version 4.0.9

- Updated package-lock.json and package.json with new Tailwind CSS version
- Added new Skeleton.astro layout with global styles and basic page structure

maribelferreira committed:
Add Footer and TopNav components with responsive design

maribelferreira committed:
Replace index.md with index.astro using Skeleton layout

maribelferreira committed:
Add favicon assets and Chakra Petch font

- Added favicon files (ico, png, webmanifest) in public/favicon
- Imported JetBrains Mono font files in public/fonts
- Updated Skeleton layout to include Google Fonts preconnect and Chakra Petch font
- Adjusted Footer component text color to use lighter gray

maribelferreira committed:
Update Footer and Homepage styling with neutral colors and content

maribelferreira committed:
Update Footer with Reddit icon SVG

maribelferreira committed:
Refactor Footer and TopNav components for improved readability and maintainability

- Extracted link arrays in Footer for cleaner template rendering
- Replaced checkbox-based mobile menu in TopNav with JavaScript toggle
- Improved semantic HTML structure and accessibility
- Added dynamic menu icon toggling with client-side script

maribelferreira committed:
Add Tailwind Typography plugin and enhance navigation styling

- Added @tailwindcss/typography plugin to package dependencies
- Updated TopNav to highlight active navigation links based on current pathname
- Added Inter font weight variants to Google Fonts link in Skeleton layout
- Refined hover states for navigation links

maribelferreira committed:
Update navigation and footer links with minor refinements

- Updated Footer component with direct GitHub prototype link and homepage logo link
- Modified navigation links in Footer (changed "/user-experience" to "/ux")
- Improved formatting and added semantic link wrapping in Footer and TopNav
- Minor adjustments to site.webmanifest

gnarea commented on src/components/Footer.astro:
In hindsight, I think we shouldn't have these two links.

gnarea commented on src/components/Footer.astro:
In hindsight, I think we shouldn't have these two links.

gnarea commented on src/components/Footer.astro:
```suggestion
          <a href="https://github.com/CheVeraId">
```

I'll move the VeraId repos from Relaycorp soon.

gnarea commented on src/components/Footer.astro:
Re: SVGs embedded in the HTML, didn't we agree to have these as separate files when we worked on awala.app?

gnarea commented on src/components/Footer.astro:
Would suggest the trick you used on awala.app to get the current year automatically.

gnarea commented on src/components/TopNav.astro:
```diff
@@ -0,0 +1,96 @@
+  </ul>
+</nav>
+
+<script>
+  document.getElementById("menu-button")?.addEventListener("click", function () {
+    const mobileMenu = document.getElementById("mobile-nav-links");
+    const menuIcon = this?.querySelector(".menu-icon");
+    const closeIcon = this?.querySelector(".close-icon");
+
+    if (mobileMenu && menuIcon && closeIcon) {
+      mobileMenu.classList.toggle("hidden");
+      menuIcon.classList.toggle("hidden");
+      closeIcon.classList.toggle("hidden");
+    }
+  });
+</script>
```
Can't we ditch this raw JS in favour of pure CSS? Please allocate a small timebox to try that, and leave it as it is it isn't straightforward.

maribelferreira replied:
Sure, I was the one who suggested it but forgot to implement it. The best pure CSS solution I found was a pure CSS using the <details> element, which gives us native toggle functionality without any JavaScript. It's semantic HTML - <details> and <summary> are meant for expandable content, and also has keyboard navigation and screen reader support.

gnarea commented on src/layouts/Skeleton.astro:
Please include the Fathom Analytics code. Check out how we did it in awala.app: https://github.com/AwalaApp/website/blob/main/src/layouts/Skeleton.astro

But in this case we need to use `data-site="UNVOWGIF"`.

gnarea reviewed (changes requested):
It's looking absolutely fantastic! Well done 💪🏾 

Apart from the in-line comments below, a couple of suggestions:

- The favicon isn't working for me on Firefox.
- The old favicon was left behind. Are you sure it's used by some browsers? If not, please delete it.

maribelferreira committed:
Add Fathom analytics in production environment

- Integrated Fathom analytics script in Skeleton layout
- Conditionally load script only in production mode
- Used Astro's environment variable detection for deployment

maribelferreira committed:
Update Footer copyright year dynamically

- Added dynamic year generation for copyright notice
- Replaced static year range with current year using JavaScript Date method

maribelferreira committed:
Add SVG icons for social media and mobile menu

- Introduced new SVG icons for X (Twitter), GitHub, Reddit, menu, and close buttons
- Replaced inline SVG icons with imported raw SVG files in Footer and TopNav
- Simplified icon rendering using Astro's set:html directive

maribelferreira committed:
Refactor TopNav mobile menu using native HTML details element

- Replaced JavaScript-based mobile menu toggle with native HTML details/summary
- Simplified mobile navigation markup and removed client-side script
- Maintained existing styling and active link highlighting
- Improved accessibility and reduced complexity of mobile menu implementation

gnarea commented on src/components/Footer.astro:
```diff
@@ -0,0 +1,77 @@
+import RedditIcon from "../assets/icons/reddit.svg?raw";
+
+const veraidLinks = [
+  { href: "/prior-art", text: "Prior art" },
+  { href: "/ux", text: "User Experience" },
+  { href: "/architecture", text: "Architecture" },
```
Sorry, I missed the fact that these don't follow the order of the top nav, and it's missing the link to the spec.

gnarea reviewed (changes requested):
Nearly there!

maribelferreira committed:
Update favicon assets and site manifest with VeraId branding

- Replaced default favicon SVG with new VeraId logo
- Updated favicon PNG and ICO files with new design
- Modified site.webmanifest to use VeraId name and correct file paths
- Adjusted favicon link tags in Skeleton layout for better browser compatibility
